app/controllers/doorkeeper/applications_controller.rb

```diff
@@ -88,10 +88,14 @@ def set_application
     end
 
     def application_params
-      params.require(:doorkeeper_application)
+      params.require(application_param_key)
         .permit(:name, :redirect_uri, :scopes, :confidential)
     end
 
+    def application_param_key
+      Doorkeeper.config.application_model.model_name.param_key
```

Comment:
This actually too strict to ActiveRecord / ActiveModel and could require additional changes for any Doorkeeper ORM (Mongoid, Sequel, etc).

Maybe it's better to use `:as` option of `form_for` helper on the application form view to set a strict parameter name (`doorkeeper_application`) ?

https://api.rubyonrails.org/v6.0.2.2/classes/ActionView/Helpers/FormHelper.html#method-i-form_for

Reply:
Yes, that's a better option. Thanks, I'll update the code.

Reply:
✔️